### DISCUSSION TITLE
explore effect of Bmark_year ranges on variance of derived quantities

### DISCUSSION THREAD

**Discussion by Rick-Methot-NOAA** [nmfs-stock-synthesis/stock-synthesis#431] [Team Posts]
While spending hours trying to find a code bug I stumbled into realizing something that I once knew and had forgotten.  The problem I saw was a change in the variance in some derived quantities. In particular, I noticed that the variance of:
#_Mgmt_Quantity
SSB_unfished 49909.7 1353.54
was less than the variance of:
#_Spawn_Bio
SSB_Virgin 49909.7 1384.54
SSB_Initial 49909.7 1384.54

and several other variances also were different because they depend on SSB_unfished.

I tracked this down to a change in Bmark_years in my input file.  If Bmark_years select just the initial equilibrium year, then variance of the two quantities are the same, but if a range of years are selected, for say biology, then the variance is reduced because the variance of a mean is always less than the variance of an individual year.  This will happen even if biology is not time-varying.  It is just basic stat 101.
Users have been routinely selecting Bmark_years without having any awareness of this effect.
I think we should create a testbed in which we demonstrate and document the effect of changing Bmark_years (and Fcast_years).

**Comment by k-doering-NOAA** [comment]
Thanks, Rick, this is helpful to know! Would the goal of the testbed here be to inform users about the potential differences in uncertainty between derived quantities? Or some other goal?

**Comment by Rick-Methot-NOAA** [comment]
Step one is to inform us!
Then inform users.
Then provide guidance on good practices or to insert warning.  For example, SS3 could determine that it is illogical to select a range of years for biology if none of the biology parameters are time-varying. 
Test would look like this:
a. create three versions of a model that estimates growth: one with no time-varying parameters, one with growth time-varying, one with selectivity time-varying.
b.  run each model with bmark_yrs all set to just start year
c.  do runs with varying combinations of ranges for Bmark_years
d.  similarly for Fcast_yrs, but here only selectivity years and relF years needs to be evaluated

**Comment by k-doering-NOAA** [comment]
Makes sense - do you mind if I open this as an issue in the ss-documentation repo?

**Comment by Rick-Methot-NOAA** [comment]
Go for it
